File: app/src/shared/entraSettings.ts
/**
 * Entra ID (Azure AD) settings expressed in the ubiquitous language of the
 * authentication domain. This is a Value Object derived from environment
 * configuration; it centralises every Entra-specific URL and identifier so no
 * other context has to hand-build them.
 */

/** The delegated permission this API requires callers to hold. */
export const REQUIRED_SCOPE = "access_as_user";

export interface EntraSettings {
  readonly tenantId: string;
  readonly clientId: string;
  readonly requiredScope: string;
  /** True once both tenant and client identifiers are present. */
  readonly isConfigured: boolean;
  /** Application ID URI that identifies this API as a resource: `api://<clientId>`. */
  readonly applicationIdUri: string;
  /** Fully-qualified delegated scope: `api://<clientId>/<scope>`. */
  readonly delegatedScopeUri: string;
  readonly authority: string;
  readonly issuer: string;
  readonly authorizationEndpoint: string;
  readonly tokenEndpoint: string;
  readonly jwksUri: string;
}

/** Load the current Entra settings from the environment. */
export function loadEntraSettings(): EntraSettings {
  const tenantId = process.env.AUTH_TENANT_ID ?? "";
  const clientId = process.env.AUTH_CLIENT_ID ?? "";
  const authority = `https://login.microsoftonline.com/${tenantId}`;

  return {
    tenantId,
    clientId,
    requiredScope: REQUIRED_SCOPE,
    isConfigured: Boolean(tenantId && clientId),
    applicationIdUri: `api://${clientId}`,
    delegatedScopeUri: `api://${clientId}/${REQUIRED_SCOPE}`,
    authority,
    issuer: `${authority}/v2.0`,
    authorizationEndpoint: `${authority}/oauth2/v2.0/authorize`,
    tokenEndpoint: `${authority}/oauth2/v2.0/token`,
    jwksUri: `${authority}/discovery/v2.0/keys`,
  };
}

File: app/src/shared/entra_settings.py
"""Entra ID (Azure AD) settings expressed in the ubiquitous language of the
authentication domain.

This is a Value Object derived from environment configuration; it centralises
every Entra-specific URL and identifier so no other context has to hand-build
them.
"""

from __future__ import annotations

import os
from dataclasses import dataclass

# The delegated permission this API requires callers to hold.
REQUIRED_SCOPE = "access_as_user"


@dataclass(frozen=True)
class EntraSettings:
    tenant_id: str
    client_id: str
    required_scope: str = REQUIRED_SCOPE

    @property
    def is_configured(self) -> bool:
        """True once both tenant and client identifiers are present."""
        return bool(self.tenant_id and self.client_id)

    @property
    def application_id_uri(self) -> str:
        """Application ID URI that identifies this API as a resource."""
        return f"api://{self.client_id}"

    @property
    def delegated_scope_uri(self) -> str:
        """Fully-qualified delegated scope: ``api://<clientId>/<scope>``."""
        return f"api://{self.client_id}/{self.required_scope}"

    @property
    def authority(self) -> str:
        return f"https://login.microsoftonline.com/{self.tenant_id}"

    @property
    def issuer(self) -> str:
        return f"{self.authority}/v2.0"

    @property
    def authorization_endpoint(self) -> str:
        return f"{self.authority}/oauth2/v2.0/authorize"

    @property
    def token_endpoint(self) -> str:
        return f"{self.authority}/oauth2/v2.0/token"

    @property
    def jwks_uri(self) -> str:
        return f"{self.authority}/discovery/v2.0/keys"


def load_entra_settings() -> EntraSettings:
    """Load the current Entra settings from the environment."""
    return EntraSettings(
        tenant_id=os.environ.get("AUTH_TENANT_ID", ""),
        client_id=os.environ.get("AUTH_CLIENT_ID", ""),
        required_scope=REQUIRED_SCOPE,
    )
